### PR TITLE
1. fix(ci): Label lwd cached state images with their sync height

### PR DESCRIPTION
**Pull request by teor2345**
## Motivation

We're currently not getting the sync height for lwd images.

## Solution

- use the lightwalletd logs for sync heights, as well as the Zebra logs
  - stop assuming all heights start with `(`
  - search back 200 lines
- add height labels to cached state images, as well as the description
- re-order lwd jobs so the full sync and update jobs come first

## Review

Anyone can review this PR.

### Reviewer Checklist

  - [ ] CI passes
  - [ ] `lwd` Images from this PR have sync heights

### CODE REVIEW

File: .github/workflows/continous-integration-docker.yml [hunk 1]
```diff
@@ -357,110 +363,115 @@ jobs:
...
       needs_zebra_state: true
-      saves_to_disk: false
+      needs_lwd_state: false
+      saves_to_disk: true
+      disk_prefix: lwd-cache
       disk_suffix: tip
       root_state_path: '/var/cache'
       zebra_state_dir: 'zebrad-cache'
+      lwd_state_dir: 'lwd-cache'
+      height_grep_text: '(current_height.*=.*Height.*\()|(Adding block to cache )'
```
Comment (teor2345) [inline .github/workflows/continous-integration-docker.yml:399]: I only added `height_grep_text` here, the rest is code movement.

File: .github/workflows/continous-integration-docker.yml [hunk 2]
```diff
@@ -357,110 +363,115 @@ jobs:
...
       needs_lwd_state: true
-      saves_to_disk: false
+      # update the disk on every PR, to increase CI speed
+      saves_to_disk: true
+      disk_prefix: lwd-cache
       disk_suffix: tip
       root_state_path: '/var/cache'
       zebra_state_dir: 'zebrad-cache'
       lwd_state_dir: 'lwd-cache'
+      height_grep_text: '(current_height.*=.*Height.*\()|(Adding block to cache )'
```
Comment (teor2345) [inline .github/workflows/continous-integration-docker.yml:427]: I only added `height_grep_text` here, the rest is code movement.

File: .github/workflows/continous-integration-docker.yml [hunk 3]
```diff
@@ -357,110 +363,115 @@ jobs:
...
-  # Test full sync of lightwalletd with a Zebra tip state
+  # Test that Zebra can answer a synthetic RPC call, using a cached Zebra tip state
   #
   # Runs:
   # - after every PR is merged to `main`
   # - on every PR update
   #
   # If the state version has changed, waits for the new cached state to be created.
   # Otherwise, if the state rebuild was skipped, runs immediately after the build job.
-  lightwalletd-full-sync:
```
Comment (teor2345) [inline .github/workflows/continous-integration-docker.yml:420]: All the rest of the changes in this file are code movement.

**Review by gustavovalverde — DISMISSED** [review]
LGTM

**Comment by teor2345** [comment]
I checked the image labels in Google Cloud Shell:
```sh
$ gcloud compute images describe lwd-cache-5086-merge-16223c7-v25-mainnet-tip-u-0906052946                                                         
archiveSizeBytes: '153503211072'
creationTimestamp: '2022-09-05T22:29:49.161-07:00'
description: Created from commit 16223c7 with height 1712961
diskSizeGb: '200'
id: '837779012590461988'
kind: compute#image
labelFingerprint: EFpi7380cG4=
labels:
  app-name: lightwalletd
  commit: 16223c7
  height: '1712961'
  network: mainnet
  purpose: lwd-cache
  state-version: '25'
  target-height: tip
  test-id: lwd-update-sync
  update-flag: -u
name: lwd-cache-5086-merge-16223c7-v25-mainnet-tip-u-0906052946
selfLink: https://www.googleapis.com/compute/v1/projects/zealous-zebra/global/images/lwd-cache-5086-merge-16223c7-v25-mainnet-tip-u-0906052946
sourceDisk: https://www.googleapis.com/compute/v1/projects/zealous-zebra/zones/us-central1-a/disks/lwd-update-sync-16223c7
sourceDiskId: '3177955771325049250'
sourceType: RAW
status: READY
storageLocations:
- us
```

**Review by gustavovalverde — APPROVED** [review]

**Pull request merged**